core/domain/question_domain.py

```diff
@@ -682,13 +682,13 @@ def _convert_state_v36_dict_to_v37_dict(cls, question_state_dict):
         Returns:
             dict. The converted question_state_dict.
         """
-        if question_state_dict['interaction']['id'] != 'TextInput':
-            return question_state_dict
-        answer_group_dicts = question_state_dict['interaction']['answer_groups']
-        for answer_group_dict in answer_group_dicts:
-            for rule_spec_dict in answer_group_dict['rule_specs']:
-                if rule_spec_dict['rule_type'] == 'CaseSensitiveEquals':
-                    rule_spec_dict['rule_type'] = 'Equals'
+        if question_state_dict['interaction']['id'] == 'TextInput':
+            answer_group_dicts = question_state_dict[
+                'interaction']['answer_groups']
+            for answer_group_dict in answer_group_dicts:
+                for rule_spec_dict in answer_group_dict['rule_specs']:
+                    if rule_spec_dict['rule_type'] == 'CaseSensitiveEquals':
+                        rule_spec_dict['rule_type'] = 'Equals'
```

Comment:
I did this just to match the pattern used everywhere else in the codebase. I also think that it is the right way and it helps to make things to the point.
Instead of returning `question_state_dict` using `question_state_dict['interaction']['id'] != 'TextInput'`.
I have made a change to run the code only when it is `question_state_dict['interaction']['id'] == 'TextInput'`.

I hope it makes sense.
ditto elsewhere (inside this file).

Reply:
I think either pattern is fine!

Reply:
@U8NWXD  So, it means we can keep these changes right?
Also, if do it the other way then I will then need to write a test for that line that does nothing. This was also one of the reasons because of which I updated it.

Reply:
Yes, the changes you made are fine with me